lib/notifications/notification.ex

```diff
@@ -27,6 +56,41 @@ defmodule Notifications.Notification do
     :operator_id,
     :operator_name,
     :route_id_at_creation,
-    :start_time
+    :start_time,
+    :end_time
   ]
+
+  @spec get_or_create(t(), (DbNotification.t() -> t())) :: t()
+  def get_or_create(notification_without_id, insertion_callback \\ fn _ -> nil end) do
+    identifying_fields =
+      Map.take(notification_without_id, [:start_time, :end_time, :block_id, :service_id, :reason])
+      |> Map.to_list()
+
+    {:ok, db_record} =
+      Skate.Repo.transaction(fn ->
+        existing_record = Skate.Repo.one(from(DbNotification, where: ^identifying_fields))
+
+        if existing_record do
+          existing_record
+        else
+          changeset =
+            DbNotification.changeset(
+              %DbNotification{},
+              Map.from_struct(notification_without_id)
+            )
+
+          db_notification = insert!(changeset)
+          notification_with_id = %{notification_without_id | id: db_notification.id}
+          log_creation(notification_with_id)
+          insertion_callback.(notification_with_id)
+          db_notification
+        end
+      end)
+
+    %__MODULE__{notification_without_id | id: db_record.id}
+  end
+
+  defp log_creation(notification) do
+    Logger.warn("Notification created new notification new_notification=#{inspect(notification)}")
```

Comment:
`Notification created new notification` sort of repeats itself, can you just say `Notification created` and drop the `new norification`? But not the `new_notification=`.